File: python/claude/p146_nre.py
# Python (Flask)
from flask import Flask, request, redirect, session
from flask_sqlalchemy import SQLAlchemy
import re

app = Flask(__name__)
app.config['SQLALCHEMY_DATABASE_URI'] = 'postgresql://user:pass@localhost/stockdb'
db = SQLAlchemy(app)

VALID_TICKERS = {'AAPL', 'GOOGL', 'MSFT', 'AMZN', 'TSLA'}
MAX_QUANTITY = 100000

@app.route('/buy', methods=['POST'])
def buy_order():
    if 'user_id' not in session:
        return redirect('/login')
    
    username = session['user_id']
    stock_name = request.form.get('stock_name', '').upper()
    stock_quantity = request.form.get('stock_quantity')
    
    if stock_name not in VALID_TICKERS:
        return 'Invalid stock', 400
    
    try:
        stock_quantity = int(stock_quantity)
        if stock_quantity <= 0 or stock_quantity > MAX_QUANTITY:
            return 'Invalid quantity', 400
    except (ValueError, TypeError):
        return 'Invalid quantity', 400
    
    try:
        db.session.execute(
            "INSERT INTO buy_orders (username, stock_name, quantity, order_date) VALUES (%s, %s, %s, NOW())",
            (username, stock_name, stock_quantity)
        )
        db.session.commit()
    except Exception as e:
        db.session.rollback()
        return 'Database error', 500
    
    return redirect('/profile')

File: javascript/claude/p146_nre.js
// JavaScript (Node.js/Express)
const express = require('express');
const session = require('express-session');
const db = require('./database');

const app = express();
app.use(express.urlencoded({ extended: false }));
app.use(session({ secret: 'key', resave: false, saveUninitialized: false }));

const VALID_TICKERS = new Set(['AAPL', 'GOOGL', 'MSFT', 'AMZN', 'TSLA']);
const MAX_QUANTITY = 100000;

app.post('/buy', async (req, res) => {
    if (!req.session.userId) {
        return res.redirect('/login');
    }
    
    const username = req.session.userId;
    const stock_name = (req.body.stock_name || '').toUpperCase();
    const stock_quantity = parseInt(req.body.stock_quantity, 10);
    
    if (!VALID_TICKERS.has(stock_name)) {
        return res.status(400).send('Invalid stock');
    }
    
    if (isNaN(stock_quantity) || stock_quantity <= 0 || stock_quantity > MAX_QUANTITY) {
        return res.status(400).send('Invalid quantity');
    }
    
    const client = await db.connect();
    try {
        await client.query('BEGIN');
        await client.query(
            'INSERT INTO buy_orders (username, stock_name, quantity, order_date) VALUES ($1, $2, $3, NOW())',
            [username, stock_name, stock_quantity]
        );
        await client.query('COMMIT');
    } catch (err) {
        await client.query('ROLLBACK');
        return res.status(500).send('Database error');
    } finally {
        client.release();
    }
    
    res.redirect('/profile');
});